src/Resource/File.php

```diff
@@ -0,0 +1,26 @@
+<?php declare(strict_types=1);
+
+namespace Becklyn\WebDav\Resource;
+
+/**
+ * @author Marko Vujnovic <mv@becklyn.com>
+ *
+ * @since  2021-08-13
+ */
+class File extends Resource
+{
+    protected function __construct(string $path, \DateTimeImmutable $lastModified, protected ?string $contentType, protected int $contentLength)
+    {
+        parent::__construct($path, $lastModified);
+    }
```

Comment:
This is something we should discuss in general: I'm not yet sure whether I'll become a huge fan of mixed argument-to-property-promotion like shown here, especially when the list tends to get longer.

I'd most likely prefer wrapping each argument in a new line, like so:

```suggestion
    protected function __construct (
        string $path, 
        \DateTimeImmutable $lastModified,
        protected ?string $contentType,
        protected int $contentLength
    )
    {
        parent::__construct($path, $lastModified);
    }
```

But then most likely staying consistent with the argument promotion and do it manually:

```suggestion
    protected function __construct (
        string $path, 
        \DateTimeImmutable $lastModified,
        ?string $contentType,
        int $contentLength
    )
    {
        parent::__construct($path, $lastModified);
        
        $this->contentType = $contentType;
        $this->contentLength = $contentLength;
    }
```

Reply:
Maybe this is a discussion for another day.

WDYT?

Reply:
Yeah I prefer 1 arg per line as well, but I often forget to do it when I have small constructors.

Regarding mixed promotion: I'm a fan of using promotion as long as we don't do anything with the props in the constructor except passing them to a parent constructor. So in this case, I'd leave the stuff as is - pass props present in the parent constructor to the parent constructor, promote the rest. But in that case I find it especially important that we have 1 arg per line, and that all of the promoted props come after the props being passed to parent constructor, otherwise it is easy to miss the props which are promoted and is generally difficult to read.

As soon as we do something with the props in the constructor other than passing them to the parent constructor, I would not promote any props and would assign all of them manually.

For example, good:
```
protected function __construct (
	string $path, 
	\DateTimeImmutable $lastModified,
	protected ?string $contentType,
	protected int $contentLength
)
{
	parent::__construct($path, $lastModified);
}
```

Bad: 
```
protected function __construct (
	string $path, 
	\DateTimeImmutable $lastModified,
	?string $contentType,
	protected int $contentLength
)
{
	parent::__construct($path, $lastModified);
        $this->foo = calcualte_foo($contentType);
}
```

The above case I'd do as:
```
protected function __construct (
	string $path, 
	\DateTimeImmutable $lastModified,
	?string $contentType,
	int $contentLength
)
{
	parent::__construct($path, $lastModified);
        $this->foo = calcualte_foo($contentType);
        $contentLength = $contentLength
}
```